src/eradiate/srf_tools.py

```diff
@@ -539,119 +536,113 @@ def threshold_filter(
     return filtered
 
 
-def integral_filter_w_bounds(
-    ds: PathLike | xr.Dataset, percentage: float = 99.0
-) -> tuple[float, float]:
-    """
-    Compute the wavelength bounds for the integral filter.
-
-    The integrated spectral reponse is computed from the cumulative sum of the
-    spectral response function interpolated on a regular grid with the
-    smallest wavelength step of the initial data set.
-
-    Parameters
-    ----------
-    ds: path-like or Dataset
-        Dataset to filter.
-
-    percentage: float
-        Keep data that contribute to this percentage of the integrated spectral
-        response.
-
-    Returns
-    -------
-    tuple
-        Wavelength bounds.
-    """
-    # convert input
-    ds = convert_no_id(ds)
+def _integral_filter_bounds_walk(
+    x: npt.ArrayLike, y: npt.ArrayLike, fraction: float
+) -> tuple[tuple[int, int], float]:
+    # Compute CDF
+    cdf = np.concatenate(([0], spint.cumulative_trapezoid(y, x)))
+    cdf /= cdf.max()
 
-    # interpolate the spectral reponse on a regular wavelength mesh
-    dwmin = ds.w.diff(dim="w").values.min()
-    wmin = ds.w.values.min()
-    wmax = ds.w.values.max()
-    wnum = int((wmax - wmin) / dwmin) + 1
-    wreg = np.linspace(wmin, wmax, wnum)
-    srfreg = ds.srf.interp(w=wreg)
+    # Compute indexes that realize the target fraction
+    i_left = np.argwhere(cdf < 0.5 * fraction).max()
+    i_right = np.argwhere(cdf > 1.0 - (0.5 * fraction)).min()
+    cs = cdf[i_right] - cdf[i_left]
 
-    # compute the cumulative sums of the interpolated spectral response
-    cumsum = np.cumsum(srfreg)
-    cumsum_max = cumsum.values.max()  # this is the integrated spectral response
+    return (i_left, i_right), cs
 
-    # this is half the fraction of data points to discard:
-    halffraction = (1 - (percentage / 100)) / 2
 
-    # wavelength points that contribute to less than half the percentage of the
-    # integrated response function, starting from the smallest wavelength values
-    w_left = cumsum.where(cumsum < halffraction * cumsum_max, drop=True).w.values
+def _integral_filter_bounds_symmetry(
+    x: npt.ArrayLike, y: npt.ArrayLike, fraction: float
+) -> tuple[tuple[int, int], float]:
+    # Insert mean in x array
+    xmean = np.trapz(y * x, x) / np.trapz(y, x)
```

Comment:
I suppose this is the mean wavelength, weighted by the SRF?

Reply:
This is $\int_a^b x f(x) \ \mathrm{d}x / \int_a^b f(x) \ \mathrm{d}x$. It's not the mean of $x f(x)$ (which would be $\frac{1}{b-a} \int_a^b f(x) \ \mathrm{d}x$). I'm actually not sure how to call it.